crates/lib3h/src/engine/network_layer.rs

```diff
@@ -215,13 +215,17 @@ impl<T: Transport, D: Dht> RealEngine<T, D> {
                     Dht::post(&mut *self.network_gateway.borrow_mut(), cmd)?;
                 } else {
                     // otherwise should be for one of our space
-                    let space_gateway = self
+                    let maybe_space_gateway = self
                         .space_gateway_map
-                        .get_mut(&(msg.space_address.to_owned(), msg.to_peer_address.to_owned()))
-                        .ok_or_else(|| {
-                            format_err!("space_gateway not found: {}", msg.space_address)
-                        })?;
-                    Dht::post(space_gateway, cmd)?;
+                        .get_mut(&(msg.space_address.to_owned(), msg.to_peer_address.to_owned()));
+                    if let Some(space_gateway) = maybe_space_gateway {
+                        Dht::post(space_gateway, cmd)?;
+                    } else {
+                        warn!(
+                            "received gossip for unjoined space_gateway: {}",
+                            msg.space_address
+                        );
+                    }
```

Comment:
We actually can receive gossip of unjoined space because other nodes are not notified when we unjoin.
The timestamp triggers this now because now when rejoining it sees an updated timestamp and so tries to gossip stuff back to the "new" peer.